spawner.py

```diff
@@ -163,9 +167,34 @@ def update_required_context(suites):
     if total == 0:
         return
 
-    # let's link to the branch overview itself
-    url = 'https://github.com/%s/commits/%s' % (os.environ['github_repo'],
-                                                os.environ['github_branch'])
+    # OK, let's upload a very basic index file that just
+    # links to the results of all the required suites
+
+    results_suites = []
+    for i, suite in enumerate(suites):
+        name = suite['context']
+        with open("state/suite-%d/url" % i) as f:
+            url = f.read().strip()
+        result = (suite['rc'] == 0)
+        results_suites.append((name, result, url))
+
+    tpl_fname = os.path.join(dirname(realpath(__file__)),
+                             'utils', 'required-index.j2')
+
+    s3_key = '%s/%s/%s.%s/%s' % (os.environ['s3_prefix'],
+                                 os.environ['github_repo'],
+                                 os.environ['github_commit'],
+                                 random.randint(0, 2**15-1),
```

Comment:
The random number (also done [in the testrunner](https://github.com/jlebon/redhat-ci/blob/08c31d0/testrunner#L124)) is a cheap way to avoid overwriting results if the same commit gets tested again (it's useful to keep the old results not only for historical reasons, but also e.g. if you have independent runs of redhat-ci against the same commit). Though maybe it's time to do this the proper way (i.e. check if it already exists rather than hope for the best).